lib/pbench/server/api/__init__.py

```diff
@@ -133,6 +134,12 @@ def register_endpoints(api: Api, app: Flask, config: PbenchServerConfig):
         endpoint="key",
         resource_class_args=(config,),
     )
+    api.add_resource(
+        Visualize,
+        f"{base_uri}/visualize/<string:dataset>",
+        endpoint="visualize",
```

Comment:
> I really don't like that form for "compare" as it makes you choose one of the arbitrary set of datasets as a special "anchor" and I don't believe that's either realistic or desirable.

Yeah, I can see that.  However, I'm pretty sure that the "baseline vs. modified" is going to be a very common form of comparison (although, then we get to argue about whether the URI parameter or the query parameter refers to the baseline 🤣).  Nevertheless, I'm dubious as to whether the multi-way comparison (as opposed to a 1-to-1) is really going to be useful or feasible -- unless we're clever, I think we're going to run out of real estate trying to do that.

At this point, I'm inclined toward making `visualize` match the existing pattern, and we can contemplate how to construct `compare` when we are closer to implementing it.

Reply:
I would expect that 2-way comparison will be common; I'm not convinced that "before vs after" is necessarily going to swamp "option A vs option B" comparisons.

Even with "before/after" we still have to be careful about defining the "primary" resource in a sub-resource API model, and I really prefer the more flexible "ordered list of resource IDs" which makes as much sense in a three-way comparison and makes the order unambiguous in a 2-way comparison.

I'm OK with changing `visualize` to a sub-resource `/datasets/<id>/visualize`, but I'm not sure I see enormous value to it either since this really is a functional API operating on a dataset rather than "sub-resource" constrained view. Or to put it another way, functionally I see more of a conceptual and functional coupling between `visualize` and `compare` than between `visualize` and `detail` or `metadata`.

Reply:
> I see more of a conceptual and functional coupling between `visualize` and `compare` than between `visualize` and `detail` or `metadata`.

If I follow that notion, then it seems like we shouldn't have separate APIs for `visualize` and `compare` -- `visualize` is just a single-result `compare`....

However, I'm thinking that `visualize` is a "show me what's in this result", just as surely as `inventory` and `contents` are, it's just that what gets displayed is nice graphics as opposed to a file list or a stream of bytes.

Reply:
> I'm not convinced that "before vs after" is necessarily going to swamp "option A vs option B" comparisons.

If someone is going to look at the comparison and say "A is X% faster than B", then I posit that B is the baseline and A is the comparison...and, I don't care whether it's "before vs. after" or "option 1 vs. option 2", one of them is going to end up in the denominator.  🙃